RocketChat/Rocket.Chat.ReactNative
# Chore: Hooks - migrate RoomHeader to hook

## Proposed changes


## Issue(s)	


## How to test or reproduce

- Open any room
- The room header should work properly

#### Test the status of another user
- Login with user A through the mobile app
- Then, on another device, login with user B
- Start a direct channel between both
- Then when user A changes his status from online, busy, away, offline, etc.
- The status in user B's RoomHeader about user A should be rendered properly

## Screenshots

## Types of changes




## Checklist



## Further comments
